.github/workflows/validate.yml

```diff
@@ -16,50 +20,52 @@ on:
 
 jobs:
   validate:
-    name: Validate ubuntu-latest ghc-${{ matrix.ghc }}
+    name: Validate ${{ matrix.os }} ghc-${{ matrix.ghc }}
 
-    runs-on: ubuntu-latest
-    continue-on-error: ${{ matrix.experimental }}
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.experimental == 'true' }}
 
     strategy:
+      # To remove after test workflow
+      fail-fast: false
       matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        ghc: ["9.2.1", "9.0.2", "9.0.1", "8.10.7", "8.8.4", "8.6.5", "8.4.4", "8.0.2"]
```

Comment:
I think you need to add `experimental: ["false"]` otherwise `experimental` might not be defined for all combinations.